File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/geocoding/AppResultLocationAdapter.java
package com.skedgo.tripkit.ui.geocoding;

import com.skedgo.tripkit.common.model.Location;
import com.skedgo.geocoding.agregator.GCAppResultInterface;
import com.skedgo.tripkit.ui.data.places.Place;
import org.jetbrains.annotations.NotNull;

public final class AppResultLocationAdapter implements ResultLocationAdapter<Place.TripGoPOI>, GCAppResultInterface {
  private final GCAppResultInterface resultInterface;
  private final Location location;

  public AppResultLocationAdapter(Location location, GCAppResultInterface resultInterface) {
    this.location = location;
    this.resultInterface = resultInterface;
  }

  @Override
  public Place.TripGoPOI getPlace() {
    return new Place.TripGoPOI(location);
  }

  @Override public String getSubtitle() {
    return resultInterface.getSubtitle();
  }

  @Override public GCAppResultInterface.Source getAppResultSource() {
    return resultInterface.getAppResultSource();
  }

  @Override public boolean isFavourite() {
    return resultInterface.isFavourite();
  }

  @NotNull
  @Override public String getName() {
    return resultInterface.getName();
  }

  @Override public Double getLat() {
    return resultInterface.getLat();
  }

  @Override public Double getLng() {
    return resultInterface.getLng();
  }
}

File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/geocoding/AppResultLocationAdapter.kt
package com.skedgo.tripkit.ui.geocoding

import com.skedgo.geocoding.agregator.GCAppResultInterface
import com.skedgo.geocoding.agregator.GCAppResultInterface.Source
import com.skedgo.tripkit.common.model.location.Location
import com.skedgo.tripkit.ui.data.places.Place.TripGoPOI

class AppResultLocationAdapter(
    private val location: Location,
    private val resultInterface: GCAppResultInterface
) : ResultLocationAdapter<TripGoPOI?>, GCAppResultInterface {
    override fun getPlace(): TripGoPOI {
        return TripGoPOI(location)
    }

    override val subtitle: String
        get() = resultInterface.subtitle

    override val appResultSource: Source
        get() = resultInterface.appResultSource

    override val isFavourite: Boolean
        get() = resultInterface.isFavourite

    override val name: String
        get() = resultInterface.name

    override val lat: Double?
        get() = resultInterface.lat

    override val lng: Double?
        get() = resultInterface.lng
}